src/theory/symmetry_detect.cpp

```diff
@@ -0,0 +1,525 @@
+/*********************                                                        */
+/*! \file symmetry_detect.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **  Paul Meng, Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Symmetry detection module
+ **/
+
+#include "theory/symmetry_detect.h"
+#include "theory/quantifiers/term_util.h"
+#include "theory/rewriter.h"
+
+using namespace std;
+
+namespace CVC4 {
+
+SymmetryDetect::Partition SymmetryDetect::detect(const vector<Node>& assertions)
+{
+  Node node = d_trueNode;
+  vector<Node>::const_iterator assertions_it = assertions.begin();
+
+  while (assertions_it != assertions.end())
+  {
+    node = NodeManager::currentNM()->mkNode(kind::AND, node, *assertions_it);
+    ++assertions_it;
+  }
+
+  Partition p = findPartitions(node);
+  Trace("sym-dt") << endl
+                  << "------------------------------ The Final Partition "
+                     "------------------------------"
+                  << endl;
+  printPartition(p);
+  return p;
+}
+
+void SymmetryDetect::getPartition(vector<vector<Node> >& parts,
+                                  const vector<Node>& assertions)
+{
+  Partition p = detect(assertions);
+  map<Node, vector<Node> >::iterator subvar_to_vars_it =
+      p.d_subvar_to_vars.begin();
+
+  while (subvar_to_vars_it != p.d_subvar_to_vars.end())
```

Comment:
It should be possible to replace this while loop with `parts.insert(p.d_subvar_to_vars.begin(), p.d_subvar_to_vars.end());`.

Reply:
Here subvar_to_vars is a map and I only need to put the value of subvar_to_vars to parts. That is why I iterated over subvar_to_vars. But I changed it to for loop.